File: 02754.py
score = [
    [4.3, 4.0, 3.7],
    [3.3, 3.0, 2.7],
    [2.3, 2.0, 1.7],
    [1.3, 1.0, 0.7]
]
grade = input()

if grade == "F":
    print('0.0')
else:
    if grade[0] == 'A':
        i = 0
    elif grade[0] == 'B':
        i = 1
    elif grade[0] == 'C':
        i = 2
    else:
        i = 3
    if grade[1] == '+':
        j = 0
    elif grade[1] == '0':
        j = 1
    else:
        j = 2

    print(score[i][j])

File: 02754.cpp
#include <iostream>
#include <string>
using namespace std;

int main() {
	double score[4][3] = { {4.3, 4.0, 3.7}, {3.3, 3.0, 2.7}, {2.3, 2.0, 1.7}, {1.3, 1.0, 0.7} };
	string grade; cin >> grade;

	if (grade == "F")
		cout << "0.0" << endl;
	else {
		int i, j;
		switch (grade[0]) {
		case 'A':
			i = 0;
			break;
		case 'B':
			i = 1;
			break;
		case 'C':
			i = 2;
			break;
		case 'D':
			i = 3;
			break;
		default:
			i = 4;
		}
		switch (grade[1]) {
		case '+':
			j = 0;
			break;
		case '0':
			j = 1;
			break;
		default:
			j = 2;
		}

		cout << fixed;
		cout.precision(1);
		cout << score[i][j] << endl;
	}
}
